mautrix_signal/signal.py

```diff
@@ -116,6 +118,38 @@ async def handle_message(self, user: 'u.User', sender: 'pu.Puppet', msg: Message
         if msg.expires_in_seconds is not None:
             await portal.update_expires_in_seconds(sender, msg.expires_in_seconds)
 
+    @staticmethod
+    async def handle_call_message(user: 'u.User', sender: 'pu.Puppet', msg: Message) -> None:
+        assert msg.call_message
+        portal = await po.Portal.get_by_chat_id(
+            sender.address, receiver=user.username, create=True
+        )
+        if not portal.mxid:
+            await portal.create_matrix_room(user, (msg.group_v2 or msg.group
+                                                   or addr_override or sender.address))
+            if not portal.mxid:
+                user.log.debug(f"Failed to create room for incoming message {msg.timestamp},"
+                               " dropping message")
+                return
+
+        msg_html = f'<a href="https://matrix.to/#/{sender.mxid}">{sender.name}</a>'
+        if msg.call_message.offer_message:
+            call_type = {
+                OfferMessageType.AUDIO_CALL: " voice ",
+                OfferMessageType.VIDEO_CALL: " video ",
+            }.get(msg.call_message.offer_message.type, " ")
+            msg_html += (
+                f" started a{call_type}call on Signal. Use the native app to answer the call."
+            )
+        elif msg.call_message.hangup_message:
+            msg_html += " ended a call on Signal."
```

Comment:
Call end notifications should be disabled by default and/or use `m.notice` instead of `m.text`